drivers/slipdev/slipdev.c

```diff
@@ -50,27 +50,89 @@ static void _slip_rx_cb(void *arg, uint8_t byte)
 {
     slipdev_t *dev = arg;
 
-    if (IS_USED(MODULE_SLIPDEV_STDIO)) {
-        if (dev->state == SLIPDEV_STATE_STDIN) {
+    switch (dev->state) {
+#if IS_USED(MODULE_SLIPDEV_STDIO)
```

Comment:
In c61a3431934a9fc3fafed11ba04d0f30 we moved the main action away from the interrupt context to increase the performance (other bytes may be missed because this callback is still going), now you are reversing that again...  Isn't there a way to do all this outside of the interrupt context, as it is in current master?

Reply:
the previous method of unescaping was rather inefficient…
Now it's just a simple `switch` - `case`.

I'm running this at 1 MHz just fine, what use case was hitting the performance limit previousy?

Reply:
:woman_shrugging: It is 5 years ago, when I did that and I only remember that the main reason were lost bytes. If it works fine at 1 MHz bautrate, I guess we can do it like this.

Reply:
I had a look at the code that was replaced by https://github.com/RIOT-OS/RIOT/commit/c61a3431934a9fc3fafed11ba04d0f30c1ce6592 it seems to me like there where more problems in slipdev then just unstuffing of bytes in the irq including the use of other parts of riot that had under/overflow issues that where solved later on. -which might lead to packages not being delivered- 
The issue connected to that bugfix #8268 #8003 has talk about some weird behavior (seems to disappear when something is printed (this indicates to me that there was something wrong in the msg handling). 

To me it seems unlikely that the unescaping in the IRQ was the cause (even though it was fixed by the same PR that removed the unescaping from the IRQ while also going for other improvements: using tsrb instead of ringbuffer (that had a underflow bug at that time), remove pktfifo[] and its cib_t (which was handled wrong).

The Board that #8003 talks about are **samr21-xpro** and reproduced on **nucleo-f446** so this might be checkable "ping from a samr21-xpro border router"